lib/cn/blk_list.c

```diff
@@ -34,18 +34,16 @@ commit_mblock(struct mpool *mp, struct kvs_block *blk)
 merr_t
 delete_mblock(struct mpool *mp, struct kvs_block *blk)
 {
-    merr_t err = 0;
-
-    if (!blk->bk_blkid)
-        return 0;
+    merr_t err;
 
     err = mpool_mblock_delete(mp, blk->bk_blkid);
     if (ev(err)) {
         log_errx("delete_mblock failed: @@e, blkid 0x%lx", err, blk->bk_blkid);
     } else {
         blk->bk_blkid = 0;
     }
-    return err;
+
+    return 0;
```

Comment:
Maybe fix the error message while you're here.  Our logging already prints the calling function name, so no need to repeat it in the message, and it's preferable that the decoded err appear that the end of the message as it does in most of our messages (???)

So, maybe something like:

log_errx("unable to delete mblock 0x%lx: @@e", err, blk->bk_blkid);

or

log_errx("mpool_mblock_delete(%p, 0x%lx) failed: @@e", err, mp, blk->bk_blkid);